### PR TITLE
DOC-2583, DOC-2846, CDS-56914

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
Added support for App Resize instance count button, support for artifact source, timeout failure strategy.

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 1]
```diff
@@ -412,7 +389,27 @@ The TAS workflow for performing a basic deployment takes your Harness TAS servic
...
...**</th> </tr> </thead> <tbody> <tr> <td>**Delegate Provisioning Errors**</td> <td>Support coming soon</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> </tr> <tr> <td>**Delegate Restart**</td> <td>Support coming soon</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> </tr> <tr> <td>**Timeout Errors**</td> <td>Support coming soon</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> </tr> <tr> <td>**Execution-time Inputs Timeout Errors**</td> <td>Support coming soon</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> <td>Rollback support required</td> </tr> </tbody> </table> |
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:405]: @shivagowda I'm assuming we need more efforts in Rollback to support this strategy. Can I mention "Support coming soon" here. (global comment)

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 2]
```diff
@@ -543,7 +581,25 @@ Use this deployment method when you want to support both new and old deployments
...
+     | **Tanzu Command** | <table> <thead> <tr> <th>Error type</th> <th>**Rollback Stage**</th> <th>**Manual Intervention**</th> <th>**Ignore Failure**</th> <th>**Retry**</th> <th>**Mark As Success**</th> <th>**Abort**</th> <th>**Mark As Failure**</th> </tr> </thead> <tbody> <tr> <td>**Delegate Provisioning Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Delegate Restart**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Timeout Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Execution-time Inputs Timeout Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> </tbody> </table> |
+
```
Comment (sourabh-awashti) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:598]: please add this line as well 
Note :- For Tanzu command step we don’t have any specific rollback step so if user wants to configure rollback then user has to write its own rollback step for this step.

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 3]
```diff
@@ -543,7 +581,25 @@ Use this deployment method when you want to support both new and old deployments
...
+     <details>
+     <summary>Error types and failure strategy</summary>
+
+     | Step name | Error types and failure strategy |
+     | :--- | :--- |
+     | **Rolling Deploy** | <table> <thead> <tr> <th>Error type</th> <th>**Rollback Stage**</th> <th>**Manual Intervention**</th> <th>**Ignore Failure**</th> <th>**Retry**</th> <th>**Mark As Success**</th> <th>**Abort**</th> <th>**Mark As Failure**</th> </tr> </thead> <tbody> <tr> <td>**Delegate Provisioning Errors**</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Delegate Restart**</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Timeout Errors**</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Execution-time Inputs Timeout Errors**</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> </tbody> </table> |
```
Comment (sourabh-awashti) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:595]: for app resize table , you can change each column from (to be supported or rollback required) to supported

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 4]
```diff
@@ -330,6 +298,14 @@ Harness services represent your microservices or applications. You can add the s
 
 1. In **Artifacts**, select **Add Artifact Source**.
 2. In **Specify Artifact Repository Type**, select **Artifactory**, and select **Continue**.
+   
+   :::important
+   TAS supports Artifactory, Nexus, Docker Registry, Amazon S3, Google Container Registry (GCR), Amazon Elastic Container Registry (ECR), Azure Container Registry (ACR), Google Artifact Registry (GAR), Google Cloud Storage (GCS), GitHub Package Registry, Azure Artifacts, Amazon Machine Image (AMI), Jenkins, Bamboo, and Custom artifact repository types.
```
Comment (RishabhGupta34) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:303]: can we remove `Custom artifact` as we removing support for that

**Review by alluvamsi-harness — DISMISSED** [review]

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md (hunk 4 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:303]: Done

**Review by sourabh-awashti — APPROVED** [review]

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 5]
```diff
@@ -42,62 +42,30 @@ You can connect Harness to a TAS space by adding a TAS connector. Perform the fo
...
-    ```
+<details>
+<summary>Install a delegate using the terminal</summary>
+<DelegateInstall />
+</details>
 
-   8. Select **Verify** to make sure that the delegate is installed properly.
-   
-   </details>
+For details and a video, go to [Delegate overview](https://developer.harness.io/docs/platform/delegates/delegate-concepts/delegate-overview).
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:68]: Suggestion:
To learn more, watch the Delegate overview video.

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 6]
```diff
@@ -330,6 +298,14 @@ Harness services represent your microservices or applications. You can add the s
 
 1. In **Artifacts**, select **Add Artifact Source**.
 2. In **Specify Artifact Repository Type**, select **Artifactory**, and select **Continue**.
+   
+   :::important
+   TAS supports Artifactory, Nexus, Docker Registry, Amazon S3, Google Container Registry (GCR), Amazon Elastic Container Registry (ECR), Azure Container Registry (ACR), Google Artifact Registry (GAR), Google Cloud Storage (GCS), GitHub Package Registry, Azure Artifacts, Amazon Machine Image (AMI), Jenkins, Bamboo repository types.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:303]:
```suggestion
   TAS supports Artifactory, Nexus, Docker Registry, Amazon S3, Google Container Registry (GCR), Amazon Elastic Container Registry (ECR), Azure Container Registry (ACR), Google Artifact Registry (GAR), Google Cloud Storage (GCS), GitHub Package Registry, Azure Artifacts, Amazon Machine Image (AMI), Jenkins, and Bamboo repository types. 
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 7]
```diff
@@ -400,9 +376,10 @@ The TAS workflow for performing a basic deployment takes your Harness TAS servic
 4. Select the **App Resize** step to define **Step Parameters**.
     1. **Name** - Edit the deployment step name.
     2. **Timeout** - Set how long you want the Harness delegate to wait for the TAS cloud to respond to API requests before timeout.
-    3. **Total Instances** - Set the number or percentage of running instances you want to keep.
-    4. **Desired Instances - Old Version** - Set the number or percentage of instances for the previous version of the application you want to keep. If this field is left empty, the desired instance count will be the difference between the maximum possible instance count (from the manifest or match running instances count) and the number of new application instances.
-    5. Select **Apply Changes**.
+    3. **Ignore instance count in Manifest** - Select this option to override the instance count mentioned in the `manifest.yaml` file with the values mentioned in the **App Resize** step.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:379]: Suggestion:
Select this option to override the instance count defined in the 'manifest.yaml' file with the values specified in the **App Resize** phase.

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 8]
```diff
@@ -400,9 +376,10 @@ The TAS workflow for performing a basic deployment takes your Harness TAS servic
...
+    4. **Total Instances** - Set the number or percentage of running instances you want to keep.
+    5. **Desired Instances - Old Version** - Set the number or percentage of instances for the previous version of the application you want to keep. If this field is left empty, the desired instance count will be the difference between the maximum possible instance count (from the manifest or match running instances count) and the number of new application instances.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:381]: Application or applications? Please consider.

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 9]
```diff
@@ -412,7 +389,31 @@ The TAS workflow for performing a basic deployment takes your Harness TAS servic
     3. Select **Apply Changes**.
    
 6. Add an **App Rollback** step to your stage if you want to roll back to an older version of the application in case of deployment failure.
-7. Select **Save**.
+7. In **Advanced** configure the following options.
+   - **Delegate Selector** - Select the delegate(s) you want to use to execute this step. You can select one or more delegates for each pipeline step. You only need to select one of a delegate's tags to select it. All delegates with the tag are selected.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:393]: Suggestions

..user for executing..
You only need to choose...

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 10]
```diff
@@ -412,7 +389,31 @@ The TAS workflow for performing a basic deployment takes your Harness TAS servic
...
+     | **Tanzu Command** | <table> <thead> <tr> <th>Error type</th> <th>**Rollback Stage**</th> <th>**Manual Intervention**</th> <th>**Ignore Failure**</th> <th>**Retry**</th> <th>**Mark As Success**</th> <th>**Abort**</th> <th>**Mark As Failure**</th> </tr> </thead> <tbody> <tr> <td>**Delegate Provisioning Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Delegate Restart**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Timeout Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Execution-time Inputs Timeout Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> </tbody> </table> |
+     
+     :::note
+     For the Tanzu Command step, we don’t provide any default rollback steps. You can do a rollback for this step by configuring your own Rollback step.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:410]:
```suggestion
     For the Tanzu Command step, Harness does not provide default rollback steps. You can do a rollback by configuring your own Rollback step.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 11]
```diff
@@ -454,7 +456,30 @@ The canary deployment contains **Canary App Setup** and **App Resize** steps. Yo
...
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:476]:
```suggestion
     For the Tanzu Command step, Harness does not provide default rollback steps. You can do a rollback for this step by configuring your own Rollback step.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 12]
```diff
@@ -515,7 +541,31 @@ Once the deployment is successful, the **Swap Routes** configuration switches th
...
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:562]:
```suggestion
     For the Tanzu Command step, Harness does not provide any default rollback steps. You can do a rollback for this step by configuring your own Rollback step.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 13]
```diff
@@ -543,7 +593,30 @@ Use this deployment method when you want to support both new and old deployments
         - **Inline** - Select this option to enter a script inline.
     3. Select **Apply Changes**.   
 5. Add a **Rolling Rollback** step to your stage if you want to rollback to an older version of the application in case of deployment failure.
-6. Select **Save**.
+6. In **Advanced** configure the following options.
+   - **Delegate Selector** - Select the delegate(s) you want to use to execute this step. You can select one or more delegates for each pipeline step. You only need to select one of a delegate's tags to select it. All delegates with the tag are selected.
+   - **Conditional Execution** - Use the conditions to determine when this step is executed. For more information, go to [conditional execution settings](/docs/continuous-delivery/x-platform-cd-features/executions/step-and-stage-conditional-execution-settings).
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:598]:
```suggestion
   - **Conditional Execution** - Use the conditions to determine when this step should be executed. For more information, go to [conditional execution settings](/docs/continuous-delivery/x-platform-cd-features/executions/step-and-stage-conditional-execution-settings).
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md [hunk 14]
```diff
@@ -543,7 +593,30 @@ Use this deployment method when you want to support both new and old deployments
...
+     | **Tanzu Command** | <table> <thead> <tr> <th>Error type</th> <th>**Rollback Stage**</th> <th>**Manual Intervention**</th> <th>**Ignore Failure**</th> <th>**Retry**</th> <th>**Mark As Success**</th> <th>**Abort**</th> <th>**Mark As Failure**</th> </tr> </thead> <tbody> <tr> <td>**Delegate Provisioning Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Delegate Restart**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Timeout Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> <tr> <td>**Execution-time Inputs Timeout Errors**</td> <td>Invalid</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> <td>Supported</td> </tr> </tbody> </table> |
+
+     
+     :::note
+     For the Tanzu Command step, we don’t provide any default rollback steps. You can do a rollback for this step by configuring your own Rollback step.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:613]:
```suggestion
     For the Tanzu Command step, Harness does not provide any default rollback steps. You can do a rollback for this step by configuring your own Rollback step.
```

**Review by SudheendraKatte — APPROVED** [review]
Approved with suggestions.

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md (hunk 8 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart.md:381]: application

**Pull request merged**